src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -242,26 +290,51 @@ export default function ActionsPanel(): JSX.Element {
   return (
     <Panel
       drawer={
-        <Button
+        <ActionButton
           appearance="positive"
-          disabled={disableSubmit}
+          loading={isExecutingAction}
+          disabled={disableSubmit || isExecutingAction}
           onClick={handleSubmit}
         >
           Run action
-        </Button>
+        </ActionButton>
       }
       width="narrow"
       data-testid={TestId.PANEL}
       title={generateTitle()}
       onRemovePanelQueryParams={handleRemovePanelQueryParams}
+      ref={scrollArea}
     >
+      <PanelInlineErrors
+        inlineErrors={
+          inlineErrors[0]
+            ? inlineErrors.map((error, index) =>
+                index === 0 ? (
```

Comment:
I think it might be nice to leave a comment here to explain why this is necessary.